src/olympia/devhub/views.py

```diff
@@ -87,12 +87,15 @@
 MDN_BASE = 'https://developer.mozilla.org/en-US/Add-ons'
 
 
-def get_fileupload_by_uuid_or_404(value):
+def get_fileupload_by_uuid_or_40x(value, *, user):
     try:
         UUID(value)
     except ValueError:
         raise http.Http404()
-    return get_object_or_404(FileUpload, uuid=value)
+    upload = get_object_or_404(FileUpload, uuid=value)
+    if upload.user != user:
+        raise PermissionDenied
+    return upload
```

Comment:
I had that initially, but figured the 403 would be easier to debug if we run into issues with this change (it has been allowed since the beginning of time...)